File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.py
def solution(num):
    cnt = 0
    if num == 1:
        answer = 0
    else:
        while num != 1:
            cnt += 1
            if num % 2 == 0:
                num //= 2
            elif num % 2 == 1:
                num = num * 3 + 1
        answer = cnt
    if cnt >= 500:
        answer = -1
    return answer

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.js
function solution(num) {
    let cnt = 0;
    while(num !== 1) {
        num = num % 2 === 0 ? num / 2 : num * 3 + 1;
        cnt ++;
        if (cnt === 500) break;
    }
    return cnt >= 500 ? -1 : cnt;
}